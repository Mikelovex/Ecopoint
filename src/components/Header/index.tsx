import { Flex, Link, Menu, MenuButton, MenuItem, MenuList, Text, Image, Button, Avatar, MenuDivider, Box, useTimeout } from '@chakra-ui/react'


import Logo from '../../assets/logo.svg'
import { useAuth } from '../../hooks/AuthContext'
import { useNavigate } from 'react-router-dom'
import { AddIcon } from '@chakra-ui/icons'
import { query, collection, where, getDocs } from 'firebase/firestore'
import { db } from '../../services/Firebase'


const Header = () => {
    const navigate = useNavigate()
    const { user, logOut } = useAuth()


    const pointsCollection = query(collection(db, 'points'), where("userId", "==", `${user.uid}`))

    const getData = async () => {
        const data = await getDocs(pointsCollection)
        const res = data.docs.map((doc) => ({ ...doc.data(), id: doc.id }))
        localStorage.removeItem('@my-points')
        localStorage.setItem('@my-points', JSON.stringify(res))
    }


    const handleLogOut = () => {
        logOut()
    }


    const handleGetPoints = async () => {
        await getData()
        setTimeout(() => {
            navigate('/company-points')
        }, 1000);
    }

    console.log('user', user)

    return (
        <Flex width="80%" height={20} margin=" 20px auto" justifyContent="space-between" alignItems={["center", "normal", "normal", "normal"]}>
            <div style={{ display: 'flex', alignItems: 'center', gap: '12px', cursor: "pointer" }} onClick={() => navigate('/')}>
                <Image width="40px" height="40px" src={Logo} alt="logo" /><h1 style={{ fontSize: '26px', fontWeight: 'bold', color: "#322153" }}>Ecopoint</h1>
            </div>
            {Object.keys(user).length > 0 ? (
                <Flex alignItems={'center'}>
                    <Link href='/faq'>
                        <Button
                            variant={'ghost'}
                            size={'sm'}
                            mr={4}
                        >
                            FAQ
                        </Button>
                    </Link>
                    <Link href='/json'>
                        <Button
                            variant={'ghost'}
                            size={'sm'}
                            mr={4}
                        >
                            JSON
                        </Button>
                    </Link>
                    <Link href='/xml'>
                        <Button
                            variant={'ghost'}
                            size={'sm'}
                            mr={4}
                        >
                            XML
                        </Button>
                    </Link>
                    <Button
                        variant={'solid'}
                        colorScheme={'blue'}
                        size={'sm'}
                        mr={4}
                        leftIcon={<AddIcon />}>
                        <Link href='/new-point'>
                            Novo ponto de coleta
                        </Link>
                    </Button>


                    <Menu>
                        <MenuButton
                            border="1px solid #3182ce"
                            as={Button}
                            rounded={'full'}
                            variant={'link'}
                            cursor={'pointer'}
                            minW={0}>
                            <Avatar
                                size={'sm'}
                                src={user.imagem}

                            />
                        </MenuButton>
                        <MenuList>
                            <Box display="flex" ml="2" mt="2" height="60px" >
                                <Avatar
                                    border="1px solid #3182ce"
                                    size={'md'}
                                    src={user.imagem}
                                />
                                <Box ml="4">
                                    <Text fontWeight="bold">{user.name}</Text>
                                    <Text mr="4" color="blackAlpha.700">{user.email}</Text>
                                </Box>
                            </Box>
                            <MenuDivider />
                            <MenuItem onClick={() => navigate("/profile")}>Perfil</MenuItem>
                            <MenuItem onClick={handleGetPoints}>Meus pontos de coleta</MenuItem>
                            <MenuDivider />
                            <MenuItem onClick={handleLogOut}>Sair</MenuItem>
                        </MenuList>
                    </Menu>
                </Flex>
            ) : (
                <Flex alignItems={'center'}>
                    <Link href='/faq'>
                        <Button
                            variant={'ghost'}
                            size={'sm'}
                            mr={4}
                        >
                            FAQ
                        </Button>
                    </Link>
                    <Link href='/json'>
                        <Button
                            variant={'ghost'}
                            size={'sm'}
                            mr={4}
                        >
                            JSON
                        </Button>
                    </Link>
                    <Link href='/xml'>
                        <Button
                            variant={'ghost'}
                            size={'sm'}
                            mr={4}
                        >
                            XML
                        </Button>
                    </Link>

                    <Link _hover={{ textDecorationLine: 'none' }} href='/sign-in'>
                        <Button
                            variant={'ghost'}
                            // colorScheme="blue"
                            size={'sm'}
                            mr={4}
                        >
                            Entrar
                        </Button>
                    </Link>
                    <Link _hover={{ textDecorationLine: 'none' }} href='/sign-up'>
                        <Button
                            variant={'solid'}
                            colorScheme="blue"
                            size={'sm'}
                            mr={4}
                        >
                            Cadastrar-se
                        </Button>
                    </Link>
                </Flex>
            )}
        </Flex >
    )
}


export default Header