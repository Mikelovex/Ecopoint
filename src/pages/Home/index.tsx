import { Button, Flex, Text, useDisclosure } from '@chakra-ui/react';
import { useEffect } from 'react'
import Header from '../../components/Header';

import { CgEnter } from 'react-icons/cg'

import BackgroundImage from '../../assets/wallpaper.svg'
import { useNavigate } from 'react-router-dom';
import { collection, doc, getDoc, getDocs, query, where } from 'firebase/firestore';
import { db } from '../../services/Firebase';
import { useAuth } from '../../hooks/AuthContext';

const Home = () => {
    const { onOpen, } = useDisclosure()
    const { user } = useAuth()
    const navigate = useNavigate()

    const pointsCollection = query(collection(db, 'points'), where("estado", "==", `São Paulo`))

    const getData = async () => {
        const data = await getDocs(pointsCollection)
        const res = data.docs.map((doc) => ({ ...doc.data(), id: doc.id }))
        localStorage.setItem('@points', JSON.stringify(res))
    }


    useEffect(() => {
        if (Object.keys(user).length > 0) {
            const docRef = doc(db, 'users', `${user.uid}`)
            getDoc(docRef).then((doc) => {
                localStorage.removeItem('@user')
                localStorage.setItem("@user", JSON.stringify(doc.data()))
            })
        }
    }, [])

    return (
        <Flex flexDirection="column" height="100vh" justifyContent="center" >
            <Header />
            <Flex maxWidth="80%" margin="auto" flexDirection="row">
                <Flex flexDirection="column" rowGap="56px" w={['100vw', '50vw', '70vw', '100vw']}>
                    <Text textAlign={['initial', 'center', 'left', 'left']} fontSize={['4xl', '3xl', '4xl', "6xl"]} fontWeight="600" color="#322153">Encontrar pontos de coleta nunca foi tão fácil.</Text>
                    <Text fontSize={["large", "2xl", "2xl", "2xl"]} color="#322153">Ajudamos pessoas a encontrarem pontos de coleta de forma eficiente</Text>
                    <Button colorScheme="blue" leftIcon={<CgEnter size={20} />} maxWidth={80} size="lg" color="white" onClick={() => {
                        localStorage.clear()
                        onOpen()
                        getData()
                        setTimeout(() => {
                            navigate('/view-points')
                        }, 2000);
                    }}>
                        Pesquisar pontos de coleta
                    </Button>
                </Flex>
                <Flex w={['0vw', '50vw', '70vw', '100vw']}>
                    <img src={BackgroundImage} alt="" />
                </Flex>
            </Flex>
        </Flex >
    )
}


export default Home;