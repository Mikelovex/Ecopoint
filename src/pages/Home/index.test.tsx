import { render, screen, waitFor } from '@testing-library/react'
import userEvent from '@testing-library/user-event'

import Home from '.'
import Header from '../../components/Header'


describe('Home Component', () => {
    test('deve conter o component de header', () => {
        render(<Home />);
        expect(screen.getByText(/Cadastre um ponto de coleta/i)).toBeInTheDocument()
    })

    test('deve conter um botão para pesquisar pontos de coleta', () => {
        render(<Home />);

        const searchButton = screen.getByRole('button', { name: /Pesquisar pontos de coleta/i })

        expect(searchButton).toBeInTheDocument()
    })

    test('deve abrir um modal ao clickar no botão de pesquisa', async () => {
        const { baseElement } = render(<Home />);

        const searchButton = screen.getByRole('button', { name: /Pesquisar pontos de coleta/i })

        userEvent.click(searchButton)

        expect(baseElement).toMatchSnapshot()

        await waitFor(() => expect(screen.queryByText('Modal Title')))
    })

    test('deve conter uma botão de pesquisa no modal', async () => {
        const { baseElement } = render(<Home />);

        const searchButton = screen.getByRole('button', { name: /Pesquisar pontos de coleta/i })

        userEvent.click(searchButton)

        expect(baseElement).toMatchSnapshot()

        await waitFor(() => expect(screen.queryByText('Buscar ponto de coleta')))
    })
})