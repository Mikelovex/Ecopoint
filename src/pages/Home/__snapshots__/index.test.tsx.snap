// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Component deve abrir um modal ao clickar no botão de pesquisa 1`] = `
<body>
  <div
    aria-hidden="true"
    data-aria-hidden="true"
    id="chakra-toast-portal"
  >
    <ul
      id="chakra-toast-manager-top"
      style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column; margin: 0px auto;"
    />
    <ul
      id="chakra-toast-manager-top-left"
      style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column;"
    />
    <ul
      id="chakra-toast-manager-top-right"
      style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column;"
    />
    <ul
      id="chakra-toast-manager-bottom-left"
      style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column;"
    />
    <ul
      id="chakra-toast-manager-bottom"
      style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column; margin: 0px auto;"
    />
    <ul
      id="chakra-toast-manager-bottom-right"
      style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column;"
    />
  </div>
  <div
    aria-hidden="true"
    data-aria-hidden="true"
  >
    <div
      class="css-fqwx7g"
    >
      <div
        class="css-7mc67y"
      >
        <img
          alt="logo"
          class="chakra-image css-ge7yul"
          src="logo.svg"
        />
        <a
          class="chakra-link css-15l38qi"
          href="new-point"
        >
          <svg
            color="#2fb86e"
            fill="none"
            height="20"
            stroke="currentColor"
            stroke-width="0"
            style="color: rgb(47, 184, 110); margin-right: 10px;"
            viewBox="0 0 24 24"
            width="20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M20 5H8V9H6V3H22V21H6V15H8V19H20V5Z"
              fill="currentColor"
            />
            <path
              d="M13.0743 16.9498L11.6601 15.5356L14.1957 13H2V11H14.1956L11.6601 8.46451L13.0743 7.05029L18.024 12L13.0743 16.9498Z"
              fill="currentColor"
            />
          </svg>
            
          Cadastre um ponto de coleta
        </a>
      </div>
      <div
        class="css-sa3qsd"
      >
        <div
          class="css-f7lypa"
        >
          <p
            class="chakra-text css-1x8bp0w"
          >
            Seu marketplace de coleta de resíduos.
          </p>
          <p
            class="chakra-text css-upeskm"
          >
            Ajudamos pessoas a encontrarem pontos de coleta de forma eficiente
          </p>
          <button
            class="chakra-button css-2rpn90"
            type="button"
          >
            <span
              class="chakra-button__icon css-1wh2kri"
            >
              <svg
                aria-hidden="true"
                fill="none"
                focusable="false"
                height="20"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 24 24"
                width="20"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M20 5H8V9H6V3H22V21H6V15H8V19H20V5Z"
                  fill="currentColor"
                />
                <path
                  d="M13.0743 16.9498L11.6601 15.5356L14.1957 13H2V11H14.1956L11.6601 8.46451L13.0743 7.05029L18.024 12L13.0743 16.9498Z"
                  fill="currentColor"
                />
              </svg>
            </span>
            Pesquisar pontos de coleta
          </button>
        </div>
        <div
          class="css-1bs7oi2"
        >
          <img
            alt=""
            src="home-background.svg"
          />
        </div>
      </div>
    </div>
  </div>
  <div
    class="chakra-portal"
  >
    <div
      aria-hidden="true"
      class="chakra-modal__overlay css-13d1tkw"
      data-aria-hidden="true"
      style="opacity: 0;"
    />
    <div
      aria-hidden="true"
      data-aria-hidden="true"
      data-focus-guard="true"
      style="width: 1px; height: 0px; padding: 0px; overflow: hidden; position: fixed; top: 1px; left: 1px;"
      tabindex="0"
    />
    <div
      aria-hidden="true"
      data-aria-hidden="true"
      data-focus-guard="true"
      style="width: 1px; height: 0px; padding: 0px; overflow: hidden; position: fixed; top: 1px; left: 1px;"
      tabindex="1"
    />
    <div
      data-focus-lock-disabled="false"
    >
      <div
        class="chakra-modal__content-container css-k2m3mz"
        tabindex="-1"
      >
        <section
          aria-describedby="chakra-modal--body-5"
          aria-labelledby="chakra-modal--header-5"
          aria-modal="true"
          class="chakra-modal__content css-6yiygt"
          id="chakra-modal-5"
          role="dialog"
          style="opacity: 0; transform: scale(0.95) translateZ(0);"
          tabindex="-1"
        >
          <header
            class="chakra-modal__header css-72fd9l"
            id="chakra-modal--header-5"
          >
            Modal Title
          </header>
          <button
            aria-label="Close"
            class="chakra-modal__close-btn css-13nquto"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="chakra-icon css-onkibi"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M.439,21.44a1.5,1.5,0,0,0,2.122,2.121L11.823,14.3a.25.25,0,0,1,.354,0l9.262,9.263a1.5,1.5,0,1,0,2.122-2.121L14.3,12.177a.25.25,0,0,1,0-.354l9.263-9.262A1.5,1.5,0,0,0,21.439.44L12.177,9.7a.25.25,0,0,1-.354,0L2.561.44A1.5,1.5,0,0,0,.439,2.561L9.7,11.823a.25.25,0,0,1,0,.354Z"
                fill="currentColor"
              />
            </svg>
          </button>
          <div
            class="chakra-modal__body css-0"
            id="chakra-modal--body-5"
          >
            <input
              class="chakra-input css-m6cbae"
              placeholder="Busque pelo nome da cidade"
              value=""
            />
          </div>
          <footer
            class="chakra-modal__footer css-192qrng"
          >
            <button
              class="chakra-button css-1warrel"
              type="button"
            >
              Buscar ponto de coleta
            </button>
          </footer>
        </section>
      </div>
    </div>
    <div
      aria-hidden="true"
      data-aria-hidden="true"
      data-focus-guard="true"
      style="width: 1px; height: 0px; padding: 0px; overflow: hidden; position: fixed; top: 1px; left: 1px;"
      tabindex="0"
    />
  </div>
</body>
`;

exports[`Home Component deve conter uma botão de pesquisa no modal 1`] = `
<body>
  <div
    aria-hidden="true"
    data-aria-hidden="true"
    id="chakra-toast-portal"
  >
    <ul
      id="chakra-toast-manager-top"
      style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column; margin: 0px auto;"
    />
    <ul
      id="chakra-toast-manager-top-left"
      style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column;"
    />
    <ul
      id="chakra-toast-manager-top-right"
      style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column;"
    />
    <ul
      id="chakra-toast-manager-bottom-left"
      style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column;"
    />
    <ul
      id="chakra-toast-manager-bottom"
      style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column; margin: 0px auto;"
    />
    <ul
      id="chakra-toast-manager-bottom-right"
      style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column;"
    />
  </div>
  <div
    aria-hidden="true"
    data-aria-hidden="true"
  >
    <div
      class="css-fqwx7g"
    >
      <div
        class="css-7mc67y"
      >
        <img
          alt="logo"
          class="chakra-image css-ge7yul"
          src="logo.svg"
        />
        <a
          class="chakra-link css-15l38qi"
          href="new-point"
        >
          <svg
            color="#2fb86e"
            fill="none"
            height="20"
            stroke="currentColor"
            stroke-width="0"
            style="color: rgb(47, 184, 110); margin-right: 10px;"
            viewBox="0 0 24 24"
            width="20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M20 5H8V9H6V3H22V21H6V15H8V19H20V5Z"
              fill="currentColor"
            />
            <path
              d="M13.0743 16.9498L11.6601 15.5356L14.1957 13H2V11H14.1956L11.6601 8.46451L13.0743 7.05029L18.024 12L13.0743 16.9498Z"
              fill="currentColor"
            />
          </svg>
            
          Cadastre um ponto de coleta
        </a>
      </div>
      <div
        class="css-sa3qsd"
      >
        <div
          class="css-f7lypa"
        >
          <p
            class="chakra-text css-1x8bp0w"
          >
            Seu marketplace de coleta de resíduos.
          </p>
          <p
            class="chakra-text css-upeskm"
          >
            Ajudamos pessoas a encontrarem pontos de coleta de forma eficiente
          </p>
          <button
            class="chakra-button css-2rpn90"
            type="button"
          >
            <span
              class="chakra-button__icon css-1wh2kri"
            >
              <svg
                aria-hidden="true"
                fill="none"
                focusable="false"
                height="20"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 24 24"
                width="20"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M20 5H8V9H6V3H22V21H6V15H8V19H20V5Z"
                  fill="currentColor"
                />
                <path
                  d="M13.0743 16.9498L11.6601 15.5356L14.1957 13H2V11H14.1956L11.6601 8.46451L13.0743 7.05029L18.024 12L13.0743 16.9498Z"
                  fill="currentColor"
                />
              </svg>
            </span>
            Pesquisar pontos de coleta
          </button>
        </div>
        <div
          class="css-1bs7oi2"
        >
          <img
            alt=""
            src="home-background.svg"
          />
        </div>
      </div>
    </div>
  </div>
  <div
    class="chakra-portal"
  >
    <div
      aria-hidden="true"
      class="chakra-modal__overlay css-13d1tkw"
      data-aria-hidden="true"
      style="opacity: 0;"
    />
    <div
      aria-hidden="true"
      data-aria-hidden="true"
      data-focus-guard="true"
      style="width: 1px; height: 0px; padding: 0px; overflow: hidden; position: fixed; top: 1px; left: 1px;"
      tabindex="0"
    />
    <div
      aria-hidden="true"
      data-aria-hidden="true"
      data-focus-guard="true"
      style="width: 1px; height: 0px; padding: 0px; overflow: hidden; position: fixed; top: 1px; left: 1px;"
      tabindex="1"
    />
    <div
      data-focus-lock-disabled="false"
    >
      <div
        class="chakra-modal__content-container css-k2m3mz"
        tabindex="-1"
      >
        <section
          aria-describedby="chakra-modal--body-7"
          aria-labelledby="chakra-modal--header-7"
          aria-modal="true"
          class="chakra-modal__content css-6yiygt"
          id="chakra-modal-7"
          role="dialog"
          style="opacity: 0; transform: scale(0.95) translateZ(0);"
          tabindex="-1"
        >
          <header
            class="chakra-modal__header css-72fd9l"
            id="chakra-modal--header-7"
          >
            Modal Title
          </header>
          <button
            aria-label="Close"
            class="chakra-modal__close-btn css-13nquto"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="chakra-icon css-onkibi"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M.439,21.44a1.5,1.5,0,0,0,2.122,2.121L11.823,14.3a.25.25,0,0,1,.354,0l9.262,9.263a1.5,1.5,0,1,0,2.122-2.121L14.3,12.177a.25.25,0,0,1,0-.354l9.263-9.262A1.5,1.5,0,0,0,21.439.44L12.177,9.7a.25.25,0,0,1-.354,0L2.561.44A1.5,1.5,0,0,0,.439,2.561L9.7,11.823a.25.25,0,0,1,0,.354Z"
                fill="currentColor"
              />
            </svg>
          </button>
          <div
            class="chakra-modal__body css-0"
            id="chakra-modal--body-7"
          >
            <input
              class="chakra-input css-m6cbae"
              placeholder="Busque pelo nome da cidade"
              value=""
            />
          </div>
          <footer
            class="chakra-modal__footer css-192qrng"
          >
            <button
              class="chakra-button css-1warrel"
              type="button"
            >
              Buscar ponto de coleta
            </button>
          </footer>
        </section>
      </div>
    </div>
    <div
      aria-hidden="true"
      data-aria-hidden="true"
      data-focus-guard="true"
      style="width: 1px; height: 0px; padding: 0px; overflow: hidden; position: fixed; top: 1px; left: 1px;"
      tabindex="0"
    />
  </div>
</body>
`;
